### PR TITLE
Datamodel

### PR DESCRIPTION
**Pull request by Feliverse**
# Blog app - creating a data model

## Description
In this project, I use migration files to build your schema.

### Project requirements
Be sure to reference the [ERD diagram](../images/blog_app_erd.png) from the sneak peek if you have any questions about table or column names or data relationships.

- Create a new app. Give it a meaningful name.
- Initialize your project with Git.
- Make sure that your project has Postgres database set up.
- Build your project schema.
    - Create and run the necessary migration files.
    - Table and column names should match the ERD diagram.
        - note: photo for users table should be a link to an image
    - Foreign keys should be included.
        - Pay attention that in `posts` table the column holding the foreign key to the `users` table should be named `author_id`
    - All columns that are foreign keys should have a corresponding index.

### Need a big picture?

Remind me about [the big picture of this project](../sneak_peek.md).

### CODE REVIEW

File: db/migrate/20230712010439_create_users.rb [hunk 1]
```diff
@@ -9,4 +9,4 @@ def change
       t.timestamps
     end
   end
-end
+end
```
Comment (CollinsTatang) [inline db/migrate/20230712010439_create_users.rb:12]: Please permit me to use this line to request changes to your README file.

- [x] Kindly fix the LICENSE link on your README file, the current link is broken.

**Review by CollinsTatang — CHANGES_REQUESTED** [review]
Hi @Feliverse, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

## Required changes 
It's ok to have bugs in your code. Please, see this code review as a chance for you to learn practical coding skills and learn new tips that can improve your code quality and help you get a better job.
_Check the comments under the review._
## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/migrate/20230712010439_create_users.rb (hunk 1 above)
Comment (Feliverse) [reply db/migrate/20230712010439_create_users.rb:12]: Thank you, I am creating the License file now.

File: db/migrate/20230712135721_create_likes.rb [hunk 2]
```diff
@@ -7,4 +7,4 @@ def change
       t.timestamps
     end
   end
-end
+end
```
Comment (Gambit142) [inline db/migrate/20230712135721_create_likes.rb:10]: - [OPTIONAL] In this [schema](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png), the `User` has many `Likes` through the column `author_id`. You need to mirror this relationship by properly configuring the migration file that creates the Likes model. I would advise you to make your migration file look this way 😎 👍:

```ruby
create_table :likes do |t|
    t.timestamps
    t.references :author, foreign_key: { to_table: 'users' }, null: false, index: true 
    t.references :post, foreign_key: true 
  end
```

File: db/migrate/20230712140459_create_comments.rb [hunk 3]
```diff
@@ -8,4 +8,4 @@ def change
       t.timestamps
     end
   end
-end
+end
```
Comment (Gambit142) [inline db/migrate/20230712140459_create_comments.rb:11]: - [OPTIONAL] In this [schema](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png), the `User` has many `Comments` through the column `author_id`. You need to mirror this relationship by properly configuring the migration file that creates the Comment model. I would advise you to make your migration file look this way 😎 👍:

```ruby
create_table :comments do |t|
    t.timestamps
    t.text :text
    t.references :author, foreign_key: { to_table: 'users' }, null: false, index: true 
    t.references :post, foreign_key: true 
  end
```

**Review by Gambit142 — APPROVED** [review]
Hi @Feliverse  👋,

## Status: Approved 🟢   

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

### Highlights
- All linter checks passed ✔️ 
- Use of `Postgresql` as database ✔️ 
- Created migration files for the `User`, `Post`, `Comment` and `Like` ✔️ 

### Optional Suggestion
- #### Kindly consider the comments tagged [OPTIONAL] under this review summary. I strongly recommend that you implement it so that you don't encounter issues in other milestones. 👍 💻 

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/migrate/20230712135721_create_likes.rb (hunk 2 above)
Comment (Feliverse) [reply db/migrate/20230712135721_create_likes.rb:10]: Thank you so much,

**Pull request merged**